lib/utilities/serverAccesssLogger.js

```diff
@@ -0,0 +1,330 @@
+}
+
+function getRequester(authInfo) {
+    const requester = null;
+    if (authInfo) {
+        if (authInfo.isRequesterPublicUser && authInfo.isRequesterPublicUser()) {
+            return requester; // Unauthenticated requests
+        } else if (authInfo.isRequesterAnIAMUser && authInfo.isRequesterAnIAMUser()) {
+            // IAM user: include IAM user name and account
+            const iamUserName = authInfo.getIAMdisplayName ? authInfo.getIAMdisplayName() : '';
+            const accountName = authInfo.getAccountDisplayName ? authInfo.getAccountDisplayName() : '';
+            return iamUserName && accountName ? `${iamUserName}:${accountName}` : authInfo.getCanonicalID();
+        } else if (authInfo.getCanonicalID) {
+            // Regular user: canonical user ID
+            return authInfo.getCanonicalID();
+        }
+    }
+    return requester;
+}
+
+function getURI(request) {
+    let requestURI = null;
+    if (request) {
+        const method = request.method || 'UNKNOWN';
+        const url = request.url || request.originalUrl || '/';
+        const httpVersion = request.httpVersion || '1.1';
+        requestURI = `${method} ${url} HTTP/${httpVersion}`;
+    }
+    return requestURI;
+}
+
+function getObjectSize(request, response) {
+    const objectSizePutMethods = Object.freeze({
+        'objectPut': true,
+        'objectPutPart': true,
+    });
+
+    const objectSizeGetMethods = Object.freeze({
+        'objectGet': true,
+    });
+
+    // If it is a PUT get the Content-Length from the request, if it is a GET get it from the response.
+    if (request && response && objectSizeGetMethods[request.apiMethod]) {
+        const len = response.getHeader('Content-Length');
+        return len || null;
+    }
+
+    if (request && objectSizePutMethods[request.apiMethod]) {
+        const len = request.headers['content-length'];
+        return len || null;
+    }
+
+    return null;
+}
+
+function getBytesSent(res, bytesSent) {
+    if (bytesSent) {
+        return bytesSent;
+    }
+
+    if (!res) {
+        return null;
+    }
+
+    const len = res.getHeader('Content-Length');
+    return len || null;
+}
+
+function calculateTotalTime(startTime, endTime) {
+    if (!startTime || !endTime) {
+        return null;
+    }
+
+    return ((endTime - startTime) / 1_000_000n).toString();
+}
+
+function calculateTurnAroundTime(startTurnAroundTime, endTurnAroundTime) {
+    if (!startTurnAroundTime || !endTurnAroundTime) {
+        return null;
+    }
+
+    return ((endTurnAroundTime - startTurnAroundTime) / 1_000_000n).toString();
+}
+
+function logServerAccess(params, req, res) {
+    const errorCode = res.serverAccessLog.errorCode;
+    const endTurnAroundTime = res.serverAccessLog.endTurnAroundTime;
+    const requestID = res.serverAccessLog.requestID;
+    const bytesSent = res.serverAccessLog.bytesSent;
+    const authInfo = params.authInfo;
+
+    serverAccessLogger.info('SERVER_ACCESS_LOG', {
```

Comment:
I think we don't use this message; we could send an empty message instead to reduce a bit the size of each log
```suggestion
    serverAccessLogger.info('', {
```

Reply:
removed